sql/core/src/main/scala/org/apache/spark/sql/execution/command/ShowNamespacesCommand.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.command
+
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.catalyst.analysis.ResolvedNamespace
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeReference}
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.util.StringUtils
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.{CatalogHelper, NamespaceHelper}
+import org.apache.spark.sql.internal.SQLConf
+import org.apache.spark.sql.types.StringType
+
+/**
+ * The command for `SHOW NAMESPACES`.
+ */
+case class ShowNamespacesCommand(
+    child: LogicalPlan,
+    pattern: Option[String],
+    override val output: Seq[Attribute] = ShowNamespacesCommand.output
+) extends UnaryRunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val ResolvedNamespace(cat, ns, _) = child
+    val nsCatalog = cat.asNamespaceCatalog
+    val namespaces = if (ns.nonEmpty) {
+      nsCatalog.listNamespaces(ns.toArray)
+    } else {
+      nsCatalog.listNamespaces()
+    }
+
+    // The legacy SHOW DATABASES command does not quote the database names.
+    val namespaceNames = if (SQLConf.get.legacyOutputSchema && namespaces.forall(_.length == 1)) {
```

Comment:
Just a question. Although `SQLConf.get.legacyOutputSchema` is `false` by default. Is this config newly applied during this migration?

Reply:
It's not new, see https://github.com/apache/spark/pull/50393/files#diff-0a770d6c621b082a7b70a93de36854206c607e94de906ce3d91ff598371cecf1L48

Reply:
BTW we should follow the previous implementation and use `output.head.name == "databaseName"` to determine if we are in legacy mode or not.

Reply:
I think we still need `assert(s.output.length == 1)` here.

Reply:
In other words, could we put the legacy check into `KeepLegacyOutputs` ?

Reply:
@beliefer  i tried this, but the new commands are in 'sql' project and is not accessible from 'catalyst'?

Reply:
Got it. It seems we can create a new KeepLegacyOutputs into `catalyst` and follows the behavior of the original one.